build.psm1

```diff
@@ -432,9 +439,10 @@ Fix steps:
     }
 
     # Framework Dependent builds do not support ReadyToRun as it needs a specific runtime to optimize for.
+    # M1/ARM64 macOS similarly does not support ReadyToRun
     # The property is set in Powershell.Common.props file.
     # We override the property through the build command line.
-    if($Options.Runtime -like 'fxdependent*' -or $ForMinimalSize) {
+    if($Options.Runtime -eq 'osx-arm64' -or $Options.Runtime -like 'fxdependent*' -or $ForMinimalSize) {
```

Comment:
I've made the change, although since your PR to move to crossgen2 is still in draft, that's probably the right place to make the change when that move happens